### DISCUSSION TITLE
CPU temperature

### DISCUSSION THREAD

**Discussion by spinau** [fastfetch-cli/fastfetch#524] [Q&A]
Has anyone had any luck getting temperature to work? I am getting -nan readout with this:--cpu-format "{1} x{3} {7}GHz ({8}º)"
(Running fastfetch on a Thinkpad P51, Archlinux 6.4.7)

**Comment by CarterLi** [comment]
Temperatures are not detected by default. Try `--cpu-temp`

**Comment by spinau** [comment]
Thanks but no luck with --cpu-temp. (I may be looking in the wrong places, but I don't even see where --cpu-temp is handled, but fastfetch does accept it.). fastfetch seems to be collecting a list of temps from my system that look like this:
detectCPUTemp name 'iwlwifi_1'
detectCPUTemp name 'thinkpad'
detectCPUTemp name 'pch_skylake'
detectCPUTemp name 'nvme'
detectCPUTemp name 'acpitz'
... none of which match the temperature names in detectCPUTemp() in cpu/cpu_linux.c, which is basically why I'm not seeing any temp printed. I'll look at this further; just got hold of the source 2.0.0.

**Reply by CarterLi** [reply]
It's in `src/detection/temps/temps_linux.c`

https://github.com/fastfetch-cli/fastfetch/blob/dev/src/detection/temps/temps_linux.c#L44

Which works for me

![image](https://github.com/fastfetch-cli/fastfetch/assets/6134068/94adadeb-900b-4526-986a-861f48e8e403)

**Comment by CarterLi** [comment]
Another option is to use thermal zone temperature

https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-class-thermal

![image](https://github.com/fastfetch-cli/fastfetch/assets/6134068/ec371518-39a4-4bdf-908d-b1087edcfb75)

Is it available for you?

**Comment by spinau** [comment]
Probably neglected to do sensors-detect and reboot. Now showing a coretemp-isa-000 (not just thinkpad-isa-000 as before). Working as advertised! Thanks for the pointers.

**Comment by CarterLi** [comment]
Strange. Mine works out of box on Fedora.
